tuf/data/roles.go

```diff
@@ -86,31 +86,139 @@ func IsDelegation(role string) bool {
 		isClean
 }
 
+// BaseRole is an internal representation of a root/targets/snapshot/timestamp role, with its public keys included
+type BaseRole struct {
+	Keys      map[string]PublicKey `json:"keys"`
```

Comment:
Non-blocking question - are these ever going to be serialized?  Should we leave out the JSON tags to make it clear that this is a storage struct, not a serialization struct?